components/admin_console/system_user_detail/team_list/team_list.jsx

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import {FormattedMessage} from 'react-intl';
+
+import {browserHistory} from 'utils/browser_history';
+import {filterAndSortTeamsByDisplayName} from 'utils/team_utils.jsx';
+
+import AbstractList from './abstract_list.jsx';
+import TeamRow from './team_row.jsx';
+
+const Header = () => (
+    <div className='groups-list--header'>
+        <div className='group-name'>
+            <FormattedMessage
+                id='admin.team_settings.team_list.nameHeader'
+                defaultMessage='Name'
+            />
+        </div>
+        <div className='group-description'>
+            <FormattedMessage
+                id='admin.systemUserDetail.teamList.header.type'
+                defaultMessage='Type'
+            />
+        </div>
+        <div className='group-description'>
+            <FormattedMessage
+                id='admin.systemUserDetail.teamList.header.role'
+                defaultMessage='Role'
+            />
+        </div>
+    </div>
+);
+
+export default class TeamList extends React.PureComponent {
+    static propTypes = {
+        userId: PropTypes.string.isRequired,
+        locale: PropTypes.string.isRequired,
+        emptyListTextId: PropTypes.string.isRequired,
+        emptyListTextDefaultMessage: PropTypes.string.isRequired,
+        actions: PropTypes.shape({
+            getTeamsData: PropTypes.func.isRequired,
+            getTeamMembersForUser: PropTypes.func.isRequired,
+        }).isRequired,
+    }
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            teamsWithMemberships: [],
+        };
+    }
+
+    componentDidMount() {
+        this.getTeamsAndMemberships().
+            then((data) => this.mergeTeamsWithMemberships(data)).
+            then((teamsWithMemberships) => {
+                this.setState(
+                    {teamsWithMemberships},
+
+                    //() => console.log(this.state.teamsWithMemberships),
+                );
+            });
+    }
+
+    getTeamsAndMemberships = async (userId = this.props.userId) => {
```

Comment:
I didn't actually know that default function parameters were evaluated at call time until I just looked it up. 👍